.yamato/scripts/update_from_upstream.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+# Prequisites:
+# This script assumes git and github CLR (gh) are installed.
+
+# Fetch from upstream and rebase our local main on it
+git remote add upstream https://github.com/dotnet/runtime
+git fetch upstream
+git checkout --track origin/main
+git rebase upstream/main
+
+# Push our rebased local main to origin
+git push -f
+
+# Create a new branch for a pull request to merge in main
+# using the current date in the branch name.
+git checkout unity-main
+branch_name=bot-upstream-main-merge-$(date '+%Y-%m-%d')
```

Comment:
Should we pass `--utc`?

Reply:
I don't think it matters too much, so long as we only do one PR per day.